test_util/test_upgrade_vpc.py

```diff
@@ -159,125 +132,277 @@ def wait_for_dns(hostname):
-    log.debug('Test app tasks at end:\n' + pprint.pformat(tasks_end))
-
-    # Verify that the tasks we started are still running.
-    assert tasks_start == tasks_end
-
-    # Verify DNS didn't fail.
-    marathon_framework_id = cluster_api.marathon.get('/v2/info').json()['frameworkId']
-    dns_app_task = cluster_api.marathon.get('/v2/apps' + dns_app['id'] + '/tasks').json()['tasks'][0]
-    dns_log = parse_dns_log(cluster_api.mesos_sandbox_file(
-        dns_app_task['slaveId'],
-        marathon_framework_id,
-        dns_app_task['id'],
-        dns_app['env']['DNS_LOG_FILENAME'],
-    ))
-    dns_failure_times = [entry[0] for entry in dns_log if entry[1] != 'SUCCESS']
-    if len(dns_failure_times) > 0:
-        message = 'Failed to resolve Marathon app hostname {} at least once.'.format(dns_app['env']['RESOLVE_NAME'])
-        log.debug(message + ' Hostname failed to resolve at these times:\n' + '\n'.join(dns_failure_times))
-        # Skip raising an exception on DNS failure so that other tests can run.
-        # DNS failure is being tracked at
-        # https://mesosphere.atlassian.net/browse/DCOS-13426.
-        #raise Exception(message)  # noqa
+
+class VpcClusterUpdateTestDcosApiSessionFactory:
+    def apply(self, dcos_url: str, masters: List[str], public_masters: List[str], slaves: List[str],
+              public_slaves: List[str], default_os_user: str) -> DcosApiSession:
+        pass
+
+
+class VpcClusterUpgradeTest:
+    log = logging.getLogger(__name__)
+
+    def __init__(self,
+                 num_masters: int, num_agents: int, num_public_agents: int,
+                 stable_installer_url: str, installer_url: str,
+                 aws_region: str, aws_access_key_id: str, aws_secret_access_key: str,
+                 ssh_key: str, default_os_user: str,
+                 config_yaml_override_install: str, config_yaml_override_upgrade: str,
+                 dcos_api_session_factory_install: VpcClusterUpdateTestDcosApiSessionFactory,
+                 dcos_api_session_factory_upgrade: VpcClusterUpdateTestDcosApiSessionFactory):
+        self.dcos_api_session_factory_install = dcos_api_session_factory_install
+        self.dcos_api_session_factory_upgrade = dcos_api_session_factory_upgrade
+        self.num_masters = num_masters
+        self.num_agents = num_agents
+        self.num_public_agents = num_public_agents
+        self.stable_installer_url = stable_installer_url
+        self.installer_url = installer_url
+        self.aws_region = aws_region
+        self.aws_access_key_id = aws_access_key_id
+        self.aws_secret_access_key = aws_secret_access_key
+        self.ssh_key = ssh_key
+        self.default_os_user = default_os_user
+        self.config_yaml_override_install = config_yaml_override_install
+        self.config_yaml_override_upgrade = config_yaml_override_upgrade
+
+        self.teamcity_msg = TeamcityServiceMessages()
```

Comment:
The pkgpanda MessageLogger doesn't currently have methods for logging test info. And since pkgpanda is not a test library I feel like it shouldn't have them added to it.